sympy/core/tests/test_evalf.py

```diff
@@ -473,3 +473,22 @@ def test_issue_9326():
     d2 = Dummy('d')
     e = d1 + d2
     assert e.evalf(subs = {d1: 1, d2: 2}) == 3
+
+def test_issue_10323_ceiling():
+    assert ceiling(Rational(2**100)+Rational(2**-100)) == 2**100+1
+    assert floor(Rational(2**100)+Rational(2**-100)) == 2**100
+    assert ceiling(Rational(2**100)-Rational(2**-100)) == 2**100
+    assert floor(Rational(2**100)-Rational(2**-100)) == 2**100-1
+
+def test_issue_10323_sqrt():
+    def asserts(x,rf,rc):
+        assert (rf == rc) == (rf**2) == x
```

Comment:
good catch on the parens, thank you.

I don't know what, if anything, besides floor/ceil of sqrt ends up calling this mess, like I said I'm new to this codebase.  Your fix may well be the right thing to do, if the need for all this logic is obsoleted by mpmath, but I am not at all sure I agree with the diagnosis as a logic expression error any longer.  It seems more like an interval arithmetic / intervals problem.

As for overkill, that's a philosophy thing.    I tend to use a lot of fuzz generators but of course you have to take overall test suite execution time into account.  

In any case my tests have other problems -- if you get past the first asserts the copysign function is undefined in the test execution context, at least when run via bin/test.

I don't see any way to withdraw the PR but I'll look at the UI again after committing this comment.